src/powerquery-language-services/analysis/analysisBase.ts

```diff
@@ -144,6 +144,23 @@ export abstract class AnalysisBase implements Analysis {
         return result ?? [];
     }
 
+    public async getFoldingRanges(): Promise<FoldingRange[]> {
```

Comment:
Depends on the cost of the operation, I guess.

Does triggering the cancellation token at the parser level cause an exception to be thrown, or does the function simply return empty results? If it's throwing, then I suppose we don't need additional handling... but otherwise, I'd assume that the language-services layer should be "cancellation aware" so it can return an empty or null result to the caller.

Reply:
A cancellation causes an error to be thrown.